src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -179,6 +184,10 @@ private String publicPathForActualPath(String actualPath, Long entityId) {
         if (actualPath.contains(Constants.LECTURE_ATTACHMENT_FILEPATH)) {
             return "/api/files/attachments/lecture/" + id + "/" + filename;
         }
+        if (actualPath.contains(Constants.FILE_UPLOAD_EXERCISES_FILEPATH)) {
+            Path path = Paths.get(actualPath);
+            return "/api/files/file-upload-exercises/" + path.getName(2) + "/submissions/" + id + "/" + filename;
```

Comment:
what is path.getName(2) --> this is bad code. Put it into a variable

Reply:
Added explicit number check and put it into a variable